File: python/easy/reverse_integer.py
def reverse(x):
    x = str(x)
    lis_x = []
    reversed_lis = []

    for i in x:
        lis_x.append(i)

    for i in reversed(x):
        reversed_lis.append(i)

    if reversed_lis[-1] == "-":
        reversed_lis.pop()
        reversed_lis.insert(0, "-")
    else:
        pass

    reversed_final = ''.join(str(i) for i in reversed_lis)
    reversed_final = int(reversed_final)

    if -2**31 <= reversed_final <= 2**31 - 1:
        print(reversed_final)
    else:
        print(0)

n = 123
reverse(n)

# n = -2**31
# print(n)
# n = 2**31 - 1
# print(n)

File: javascript/easy/reverse_integer.js
// 7. Reverse Integer
// https://leetcode.com/problems/reverse-integer/

var reverse = function(x) {
    var x = x.toString();
    var list_x = x.split('');
    var reversed_list = list_x.reverse();

    var last_element = reversed_list[reversed_list.length - 1];

    if (last_element == "-") {
        reversed_list.pop();
        reversed_list.unshift("-");
    }

    var result = reversed_list.join('');
    var final = parseInt(result);

    const upper_bounds = Math.pow(2, 31) - 1;
    const lower_bounds = Math.pow(2, 31) * -1;

    if (final < lower_bounds || final > upper_bounds) {
        return 0;
    } else {
        return final;
    }
}